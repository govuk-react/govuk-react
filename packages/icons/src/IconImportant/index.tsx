import React from 'react';

import SVG, { SVGProps } from '../SVGBase/index';

interface IconImportantProps extends SVGProps {
  title?: string;
  fill?: string;
}

const IconImportant = ({ fill, title, ...rest }: IconImportantProps) => (
  <SVG viewBox="0 0 35.000000 35.000000" preserveAspectRatio="xMidYMid meet" title={title} fill={fill} {...rest}>
    <g transform="translate(0.000000,35.000000) scale(0.100000,-0.100000)">
      <path
        d="M100 332 c-87 -48 -125 -155 -82 -232 48 -87 155 -125 232 -82 87 48
125 155 82 232 -48 87 -155 125 -232 82z m100 -122 c0 -53 -2 -60 -20 -60 -18
0 -20 7 -20 60 0 53 2 60 20 60 18 0 20 -7 20 -60z m0 -111 c0 -12 -7 -19 -20
-19 -19 0 -28 28 -14 43 11 11 34 -5 34 -24z"
      />
    </g>
  </SVG>
);

IconImportant.defaultProps = {
  title: 'icon important',
  fill: 'currentColor',
};

export default IconImportant;
