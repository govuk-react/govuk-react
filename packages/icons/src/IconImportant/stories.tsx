import type { Story } from '@storybook/react';

import * as React from 'react';

import IconImportantComponent from '.';

export default {
  title: 'Utility/Icons/IconImportant',
};

export const IconImportant: Story = (args) => {
  return <IconImportantComponent {...args} title="Example IconImportant implementation" />;
};
IconImportant.args = {
  width: '256px',
  height: '256px',
  fill: 'black',
};
