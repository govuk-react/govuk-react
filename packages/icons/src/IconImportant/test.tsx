import * as React from 'react';
import ReactDOM from 'react-dom';
import { mount } from 'enzyme';
import IconImportant from '.';

const wrapper = <IconImportant>Example</IconImportant>;

describe('IconImportant', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(wrapper, div);
  });

  it('matches wrapper snapshot', () => {
    expect(mount(wrapper)).toMatchSnapshot('wrapper mount');
  });
});
