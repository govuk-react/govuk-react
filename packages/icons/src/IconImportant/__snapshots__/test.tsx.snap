// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconImportant matches wrapper snapshot: wrapper mount 1`] = `
.c0 {
  display: block;
}

<IconImportant
  fill="currentColor"
  title="icon important"
>
  <SVG
    fill="currentColor"
    preserveAspectRatio="xMidYMid meet"
    title="icon important"
    viewBox="0 0 35.000000 35.000000"
    width="100%"
  >
    <SVGBase__StyledSvg
      fill="currentColor"
      height="100%"
      preserveAspectRatio="xMidYMid meet"
      version="1.1"
      viewBox="0 0 35.000000 35.000000"
      width="100%"
    >
      <svg
        className="c0"
        fill="currentColor"
        height="100%"
        preserveAspectRatio="xMidYMid meet"
        version="1.1"
        viewBox="0 0 35.000000 35.000000"
        width="100%"
      >
        <title>
          icon important
        </title>
        <g
          transform="translate(0.000000,35.000000) scale(0.100000,-0.100000)"
        >
          <path
            d="M100 332 c-87 -48 -125 -155 -82 -232 48 -87 155 -125 232 -82 87 48
125 155 82 232 -48 87 -155 125 -232 82z m100 -122 c0 -53 -2 -60 -20 -60 -18
0 -20 7 -20 60 0 53 2 60 20 60 18 0 20 -7 20 -60z m0 -111 c0 -12 -7 -19 -20
-19 -19 0 -28 28 -14 43 11 11 34 -5 34 -24z"
          />
        </g>
      </svg>
    </SVGBase__StyledSvg>
  </SVG>
</IconImportant>
`;
